SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DefaultApplicationSettings.cs

```diff
@@ -51,33 +83,56 @@ public bool ContainsKey(string key)
         {
             lock (configLock)
             {
-                return configuration.AppSettings.Settings[key] != null;
+                if (configuration != null)
+                {
+                    return configuration.AppSettings.Settings[key] != null;
+                }
+                else
+                {
+                    AppCenterLog.Warn(AppCenterLog.LogTag, CorruptedConfigurationWarning);
+                }
             }
+
+            return false;
         }
 
         public void Remove(string key)
         {
             lock (configLock)
             {
-                configuration.AppSettings.Settings.Remove(key);
-                SaveConfiguration();
+                if (configuration != null)
+                {
+                    configuration.AppSettings.Settings.Remove(key);
+                    SaveConfiguration();
+                }
+                else
+                {
+                    AppCenterLog.Warn(AppCenterLog.LogTag, CorruptedConfigurationWarning);
+                }
             }
         }
 
         private void SaveValue(string key, string value)
         {
             lock (configLock)
             {
-                var element = configuration.AppSettings.Settings[key];
-                if (element == null)
+                if (configuration != null)
                 {
-                    configuration.AppSettings.Settings.Add(key, value);
+                    var element = configuration.AppSettings.Settings[key];
+                    if (element == null)
+                    {
+                        configuration.AppSettings.Settings.Add(key, value);
+                    }
+                    else
+                    {
+                        element.Value = value;
+                    }
+                    SaveConfiguration();
                 }
                 else
                 {
-                    element.Value = value;
+                    AppCenterLog.Warn(AppCenterLog.LogTag, CorruptedConfigurationWarning);
```

Comment:
Can we log what has failed to save, at least the key? I'm just looking from developer perspective - how we are going to debug this.